### PR TITLE
Remove redundant third party dependencies.

### PR DESCRIPTION
**Pull request by johnamcleod**

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -53,13 +48,6 @@ taskipy = "^1.2.0"
 pytest-black = "^0.3.11"
 black = { version = "*", allow-prereleases = true }
 pytest-isort = "^1.0.0"
-sphinx = "^3.2.1"
```
Comment (pvrancx) [inline pyproject.toml:56]: looks like you're removing the docs libraries - are there instructions somewhere how to install these if you want to build the docs?

**Review by pvrancx — DISMISSED** [review]
assuming install still works - LGTM  ;-)

File: pyproject.toml (hunk 1 above)
Comment (johnamcleod) [reply pyproject.toml:56]: The docs requirements were moved into the `docs` package and should have been removed from here: https://github.com/Bellman-devs/bellman/blob/develop/docs/docs_requirements.txt

This PR is catching up with that task.

Comment (hstojic) [reply pyproject.toml:56]: we need to leave these in here, we use poetry to build docs as well and buidling docs in general I would consider develoment dependencies, so they should stay

**Review by hstojic — COMMENTED** [review]
can we discuss first that removal of docs dependencies from development section of pyroject.toml file?

**Comment by pvrancx** [comment]
> can we discuss first that removal of docs dependencies from development section of pyroject.toml file?

I'm ok with docs having a separate build process, provided there is an automated command and it is documented.
I could see this being a part of the dev install, but don't feel strongly about this. TBH I tend not to build docs and just consult them online.

**Comment by johnamcleod** [comment]
> > can we discuss first that removal of docs dependencies from development section of pyroject.toml file?
> 
> I'm ok with docs having a separate build process, provided there is an automated command and it is documented.
> I could see this being a part of the dev install, but don't feel strongly about this. TBH I tend not to build docs and just consult them online.

@hstojic is right - I had forgotten that there is a poetry command to build the docs, which at the moment does not install anything. There are three ways to address this:
1) update the command which builds the docs to install the extra requirements
2) Restore the dev dependencies for building the docs
3) Restore the previous action definition which explicitly installed the extra requirements

I would prefer the keep the dev dependencies and doc dependencies separate, so I would prefer 1) or 3), but I do not feel strongly about this.

**Comment by pvrancx** [comment]
> * update the command which builds the docs to install the extra requirements
> * Restore the dev dependencies for building the docs
> * Restore the previous action definition which explicitly installed the extra requirements

So 1) would enable a separate poetry command that installs the dependencies and builds the docs? That would be my preference. I'd prefer to minimize the dependencies you are forced to install. Having a separate command gives maximum control to the user.

**Comment by hstojic** [comment]
not sure I understand what is 3)...

what about 4) where in dev part  of dependencies we specify docs related packages as optional (same as for mujoco-py), for those contributors that might work on docs and will need/want to check whether it builds?

that would be better than 1) which would install packages every time I build docs (which might happen a lot when  you are actively working on them)

depending on what we implement, we should update `CONTRIBUTING.md`, where it talks about building docs...

**Comment by johnamcleod** [comment]
> what about 4) where in dev part of dependencies we specify docs related packages as optional (same as for mujoco-py), for those contributors that might work on docs and will need/want to check whether it builds?

That's good - I will go with that.

> depending on what we implement, we should update `CONTRIBUTING.md`, where it talks about building docs...

Good point - I will also update this.

**Comment by hstojic** [comment]
4) doesn't seem to work, just tried it

it seems that optional flag cannot be used in dev-dependencies block: https://stackoverflow.com/questions/60971502/python-poetry-how-to-install-optional-dependencies#60990574

I think I would then go with 2), only contributors will be installing these few additional libs, and if they are using poetry then they are installing them into a virtual envirionment and there shouldnt be any issue with installing these few libs for the project...

**Comment by johnamcleod** [comment]
I have updated the contributors file to describe how to build the docs, and I have verified locally that the instructions work. The documentation build dependencies are listed in the `docs_requirements.txt` file, which can be installed separately from the dev dependencies to build the docs.

**Review by pvrancx — APPROVED** [review]
Looks fine to me - does this work for you @hstojic ?

**Pull request merged**